### ISSUE TITLE
Implement "Last Changed Rev" instead of "Revision" field

### ISSUE THREAD

**Issue by moreaki** [mojohaus/buildnumber-maven-plugin#66]
It seems that the current notion of picking the revision information is the `Revision` field from the `svn info` information. However, I would need to have the `Last Changed Rev` information field. After searching the web and reading the documentation, I have not found a way to obtain this field. Is there a way, and if not, would it be possible to implement this (along the line of https://github.com/mojohaus/buildnumber-maven-plugin/pull/16/commits/8c0f41458e55e2fe3e3e9644437f7e1c832346e9)?

Example output from `svn info`:

``` yaml
Path: .
Working Copy Root Path: /<...>/dev/projects/proj-A
URL: http://example.com/svn/<...>/trunk
Relative URL: ^/<...>/trunk
Repository Root: http://example.com/svn/
Repository UUID: 8104d0a4-ed15-438a-b3ec-15d70441338f
Revision: 78855
Node Kind: directory
Schedule: normal
Last Changed Author: the.swede
Last Changed Rev: 78847
Last Changed Date: 2017-06-28 09:43:58 +0900 (Wed, 28 Jun 2017)
```

In the above excerpt one can spot that the overall company-wide svn revision counter field is at **78855**, while the individual project's svn counter for the last change is at **78847**. This is the one I would need.

Quite related is the following question on [StackOverflow](https://stackoverflow.com/questions/40438351/retrieve-svn-last-changed-revision-number-with-the-buildnumber-maven-plugin).

**Comment by moreaki** [comment]
Looking briefly at the source code, I notice that there is an `InfoItem` class, which essentially seems to implement just that (judging by its private fields):

```java
/**
 * @author <a href="mailto:kenney@apache.org">Kenney Westerhof</a>
 * @author Olivier Lamy
 *
 * @since 1.5
 */
public class InfoItem
{
    private String path;
    private String url;
    private String repositoryRoot;
    private String repositoryUUID;
    private String revision;
    private String nodeKind;
    private String schedule;
    private String lastChangedAuthor;
    private String lastChangedRevision;
    private String lastChangedDate;
```

It also says since 1.5, which I cannot find yet released.

When looking at the `AbstractScmMojo`, I find the following code responsible to fetch the last revision:

```java
    protected String getScmRevision()
        throws ScmException
    {
        ScmRepository repository = getScmRepository();

        InfoScmResult scmResult = info( repository, new ScmFileSet( scmDirectory ) );

        if ( scmResult == null || scmResult.getInfoItems().isEmpty() )
        {
            return ( !StringUtils.isEmpty( revisionOnScmFailure ) ) ? revisionOnScmFailure : null;
        }

        checkResult( scmResult );

        InfoItem info = scmResult.getInfoItems().get( 0 );

        if ( useLastCommittedRevision )
        {
            // It seemed the case that for git getLastChangedRevision()
            // returns null instead of the last revision in contradiction
            // to the above code: scmResult.getInfoItems().get(0);
            return info.getLastChangedRevision();
        }

        return info.getRevision();
    }
```

Could it be possible that issue this is hitting me?

**Comment by Dacesilian** [comment]
Is there any progress on it? I need to get SVN commit date, not build date. Thanks.

**Comment by moreaki** [comment]
Nope, no progress has been made, and from the commit log (1 commit since this issue ten months ago) it does not seem like there is much development going on either in for the project. It's unfortunate because for the original authors it would probably be very easy to address.

**Comment by khmarbaise** [comment]
What about making a branch and integrating the code changes you think are responsible and simply test it?

**Comment by glucas** [comment]
This appears to be working if I build the plugin locally and use the 3.0.0-SNAPSHOT version rather than the released 1.4 version.
